Source/TCPConnection.swift

```diff
@@ -30,145 +30,97 @@ public final class TCPConnection: Connection {
     var socket: tcpsock?
     public private(set) var closed = true
 
-    init(with socket: tcpsock) throws {
-        self.ip = try IP(address: tcpaddr(socket))
+    internal init(with socket: tcpsock) throws {
+        let address = tcpaddr(socket)
+        try ensureLastOperationSucceeded()
+        self.ip = IP(address: address)
         self.socket = socket
         self.closed = false
-        try assertNotClosed()
     }
 
     public init(to host: String, on port: Int) throws {
-        if host == "0.0.0.0" || host == "127.0.0.1" {
-            self.ip = try IP(localAddress: host, port: port)
-        } else {
-            self.ip = try IP(remoteAddress: host, port: port)
-        }
+        self.ip = try IP(remoteAddress: host, port: port)
```

Comment:
Are you sure about that? I think `IP(remoteAddress: host, port: port)` tries to resolve locally if it fails, then it tries remotely.